video_xblock/static/js/video_xblock.js

```diff
@@ -46,12 +47,23 @@ function VideoXBlockStudentViewInit(runtime, element) {
       // Discard a message received from another domain
       return;
     try {
+      if (event.data.action === "saveState") {
+        updateTranscriptDownloadUrl(event.data.downloadTranscriptUrl);
+      };
+
       var url = handlers[event.data.action][event.data.xblockUsageId];
       if (url) {
         sendData(url, event.data.info);
       }
     } catch (err){
       console.log(err)
     }
+  };
+  /** Updates transcript download url if it is enabled */
+  function updateTranscriptDownloadUrl(downloadTranscriptUrl) {
+    try {
+      var downloadLinkEl = document.getElementById('download-transcript-link');
+      downloadLinkEl.href = downloadTranscriptHandlerUrl + '?' + downloadTranscriptUrl;
+    } catch (err){}
```

Comment:
When download transcript is disabled in video xblock settings, downloadLinkEl is equal to null, in this case in line 66 we get this error
TypeError: Cannot set property 'href' of null